app/api/groups/update_group.go

```diff
@@ -141,6 +149,51 @@ func (srv *Service) updateGroup(w http.ResponseWriter, r *http.Request) service.
 	return service.NoError
 }
 
+func validateActivityIDAndIsOfficial(
+	store *database.DataStore, user *database.User, oldActivityID *int64, oldIsOfficial bool,
+	dbMap map[string]interface{}) service.APIError {
+	activityIDToCheck := oldActivityID
+	activityID, activityIDSet := dbMap["activity_id"]
+	if activityIDSet {
+		activityIDToCheck = activityID.(*int64)
+		apiError := validateActivityID(store, user, activityIDToCheck)
+		if apiError != service.NoError {
+			return apiError
+		}
+	}
+
+	isOfficialSession, isOfficialSessionSet := dbMap["is_official_session"]
+	if (isOfficialSessionSet && isOfficialSession.(bool)) || (!isOfficialSessionSet && oldIsOfficial && activityIDSet) {
+		if activityIDToCheck == nil {
+			return service.ErrInvalidRequest(errors.New("the activity_id should be set for official sessions"))
+		}
+		found, err := store.PermissionsGranted().WithWriteLock().
+			Joins(`
+				JOIN groups_ancestors ON groups_ancestors.ancestor_group_id = permissions_granted.group_id AND
+				     groups_ancestors.child_group_id = ?`, user.GroupID).
+			Where("permissions_granted.can_make_session_official").
+			Where("permissions_granted.item_id = ?", activityIDToCheck).
+			HasRows()
+		service.MustNotBeError(err)
+		if !found {
+			return service.ErrForbidden(errors.New("not enough permissions for attaching the group to the activity as an official session"))
+		}
+	}
+	return service.NoError
+}
```

Comment:
There is something that may be annoying, even if probably already like that in many other places:
If the activity_id is given but the same as before, possibly it is just the former data which is resubmitted as is (just changing the name for instance)... but so in this case the user is not setting a new activity, and so it should not be rechecked as he might not have the perm while his colleague who set it earlier had the perm. 
So IMO, `activityIDSet`/isOfficialSessionSet are not what should be used but rather, `activityIDChanged`/`isOfficialSessionChanged`

Reply:
fixed